### PR TITLE
Target link libraries

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**

### CODE REVIEW

File: clang/test/dpct/cmake_migation/case_004/case_004_target_link_libraries.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: rm -rf %T && mkdir -p %T
+// RUN: cd %T
+// RUN: cp %S/input.cmake ./input.cmake
+// RUN: dpct -in-root ./ -out-root out  ./input.cmake --rule-file=%S/../../../../tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml --migrate-cmake-script-only --cuda-include-path="%cuda-path/include"
```
Comment (tomflinda) [inline clang/test/dpct/cmake_migation/case_004/case_004_target_link_libraries.cpp]: --rule-file=%S/../../../../tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml and --cuda-include-path="%cuda-path/include" could be removed.

File: clang/test/dpct/cmake_migation/case_004/input.cmake [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Project(foo-bar LANGUAGES CXX CUDA)
```
Comment (tomflinda) [inline clang/test/dpct/cmake_migation/case_004/input.cmake]: Line1 is not related to this lit test, pls remove it.

File: clang/tools/dpct/cmake/dpct.cmake [hunk 3]
```diff
@@ -75,6 +75,40 @@ macro(DPCT_COMPILE_SYCL_CODE generated_files)
   DPCT_COMPILE_SYCL_CODE_IMP(sycl_device ${generated_files} ${ARGN})
 endmacro()
 
+macro (target_link_sycl_libraries target link_libraries_keyword)
```
Comment (tomflinda) [inline clang/tools/dpct/cmake/dpct.cmake:82]: Pls add behavior test for  target_link_sycl_libraries in https://github.com/oneapi-src/SYCLomatic-test

File: clang/test/dpct/cmake_migation/case_004/case_004_target_link_libraries.cpp (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/test/dpct/cmake_migation/case_004/case_004_target_link_libraries.cpp]: Done.

**Pull request closed without merging**